prover/src/handlers.rs

```diff
@@ -64,13 +64,25 @@ pub async fn prove_handler(
     let input = preprocess::decode_and_add_jwk(body, jwk_override.as_ref())
         .with_status(StatusCode::BAD_REQUEST)?;
 
-    training_wheels::check_nonce_consistency(&input, &state.circuit_config)
+    #[allow(clippy::match_like_matches_macro)]
+    let use_new_setup = match (
+        ON_CHAIN_GROTH16_VK.read().unwrap().as_ref(),
+        state.new_groth16_vk.as_ref(),
+    ) {
+        (Some(on_chain), Some(local)) if on_chain == local => true,
+        _ => false,
```

Comment:
Under what circumstances would `state.new_groth16_vk.as_ref()` return `None`? Misconfiguration?

If so, returning `false` deals with this well?

(I see how `ON_CHAIN_GROTH16_VK` could be `None` due to misconfiguration.)

Reply:
`state.new_groth16_vk` can be none mainly when new vk is not specified.
`ON_CHAIN_GROTH16_VK` can be none mainly when prover just starts and the on chain value hasn't been fetched.